tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+tmp_csv = '%s.tmp.csv' % (fname_in)
+fname_out = '%s.features.csv' % fname_in
+t_fname_out = '%s.features.transposed.csv' % fname_in
+
+
+class TestTsfreshBaseline(unittest.TestCase):
+    """
+    Test all the features and their calculated values with a 60 data point
+    sample of simple anomalous timeseries data set and compare that the feature
+    names and calculated values match the baselines calcualated for the specific
+    verison of tsfresh as defined by :mod:`tsfresh_feature_names.TSFRESH_FEATURES`
+
+    .. warning:: the Python 2 and 3 calculate different results in terms of
+        float precision therefore baseline transposed features csv files are
+        required for both py2 and py3.
+
+    Running the test
+    ================
+
+    .. code-block:: bash
+
+        cd "<YOUR_TSFRESH_DIR>"
+        python -m pytest tests/baseline/tsfresh_features_test.py
+
+
+    Test the test fails
+    ===================
+
+    To test that the test fails as desired and as does what it is supposed to do
+    there are 2 methods to achieve this:
+
+    - Modify the first value in your local tsfresh/examples/data/test_tsfresh_baseline_dataset/data.json
+      and run the test, then delete the modified local data.json file to be pulled down again.
+    - Modify a feature name or value in your local tests/baseline/tsfresh-<TSFRESH_BASELINE_VERSION>.py<PYTHON_VERSION>.data.json.features.transposed.csv file,
+      run the test and either pull it again or revert the change
+
+    """
+    def setUp(self):
+        self.baseline_ts_json = download_json_dataset()
+
+    def tearDown(self):
+        # Remove the directory after the test
+        try:
+            shutil.rmtree(test_path)
+        except:
+            pass
+
+    def test_tsfresh_baseline(self):
+        """
+        This test should generates a set of resources from a sample 60 data
+        point timeseries, extracts the features and saves a transposed csv of
+        feature_name,value that is used to compare against to baseline
+        transposed csv
+        """
+        baseline_ts_json = self.baseline_ts_json
+
+        test_path_created = False
+        if os.path.exists(test_path):
+            test_path_created = True
+        self.assertEqual(test_path_created, True)
+
+        json_data_exists = False
+        if os.path.isfile(baseline_ts_json):
+            json_data_exists = True
+
+        self.assertEqual(json_data_exists, True)
+
+        timeseries_json = None
+        if os.path.isfile(baseline_ts_json):
+            with open(baseline_ts_json, 'r') as f:
+                timeseries_json = json.loads(f.read())
+
+        if python_version == 2:
+            timeseries_str = str(timeseries_json).replace('{u\'results\': ', '').replace('}', '')
+        if python_version == 3:
+            timeseries_str = str(timeseries_json).replace('{\'results\': ', '').replace('}', '')
+
+        full_timeseries = eval(timeseries_str)
```

Comment:
Changed.  Seeing as it was my data source I trusted it :)  However, point taken, you would think Python would just deprecate eval or issue a user warning, "try use  ast.literal_eval here" or the linter with some PEP warning :)